Realm/RLMSyncUser.mm

```diff
@@ -253,6 +254,7 @@ + (void)_performLogInForUser:(RLMSyncUser *)user
                     return;
                 }
                 user->_user = sync_user;
+                user.isAdmin = model.refreshToken.tokenData.isAdmin;
```

Comment:
ISTM that the concepts are really:
* A user can be an admin or a normal user. Distinguishing those is what this PR is about.
* The initial admin user (I think? Or is it more than just the initial user?) can authenticate via a special access token, bypassing the normal authentication flow. This is what `SyncUser`'s `m_is_admin` currently tracks. This concept should probably be renamed to make it clear that it's about how the authentication is handled.